BudgetDebitor fails to claim budget after sometime

```
org.agrona.concurrent.AgentTerminationException: java.lang.IllegalStateException: missing file for budgets : /Users/akramyakubov/workspace/aklivity/local-zilla/.zilla/engine/budgets127
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:707)
	at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
	at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.IllegalStateException: missing file for budgets : /Users/akramyakubov/workspace/aklivity/local-zilla/.zilla/engine/budgets127
	at org.agrona.core/org.agrona.IoUtil.checkFileExists(IoUtil.java:507)
	at org.agrona.core/org.agrona.IoUtil.mapExistingFile(IoUtil.java:396)
	at org.agrona.core/org.agrona.IoUtil.mapExistingFile(IoUtil.java:361)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.layouts.BudgetsLayout$Builder.build(BudgetsLayout.java:144)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.newBudgetDebitor(DispatchAgent.java:1620)
	at org.agrona.core/org.agrona.collections.Int2ObjectHashMap.computeIfAbsent(Int2ObjectHashMap.java:268)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.supplyDebitor(DispatchAgent.java:536)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.budget.KafkaMergedBudgetAccountant.supplyDebitor(KafkaMergedBudgetAccountant.java:49)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$ClusterClient.onNetworkWindow(KafkaClientGroupFactory.java:1890)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$ClusterClient.onNetwork(KafkaClientGroupFactory.java:1757)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool.doWindow(KafkaClientConnectionPool.java:424)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.doStreamWindow(KafkaClientConnectionPool.java:805)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.doStreamBegin(KafkaClientConnectionPool.java:818)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.onStreamBegin(KafkaClientConnectionPool.java:668)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.onStreamSignal(KafkaClientConnectionPool.java:992)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientConnection.onConnectionSignal(KafkaClientConnectionPool.java:1466)
	at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientConnection.onConnectionMessage(KafkaClientConnectionPool.java:1327)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1119)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1041)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
	at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:701)
	... 3 more
	Suppressed: java.lang.Exception: [engine/data#9]	[0x0909000000000231] streams=[consumeAt=0x0002eec8 (0x000000000082eec8), produceAt=0x0002efe0 (0x000000000082efe0)]
		at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:705)
		... 3 more
stopped
┌─(~/workspace/ak
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -17,7 +17,6 @@ package io.aklivity.zilla.runtime.binding.kafka.internal.stream;
 
 import static io.aklivity.zilla.runtime.binding.kafka.internal.types.ProxyAddressProtocol.STREAM;
 import static io.aklivity.zilla.runtime.engine.budget.BudgetCreditor.NO_BUDGET_ID;
-import static io.aklivity.zilla.runtime.engine.budget.BudgetCreditor.NO_CREDITOR_INDEX;
 import static io.aklivity.zilla.runtime.engine.concurrent.Signaler.NO_CANCEL_ID;
 import static java.lang.System.currentTimeMillis;
 
@@ -666,6 +665,7 @@ public final class KafkaClientConnectionPool
             state = KafkaState.openingInitial(state);
 
             doStreamBegin(authorization, traceId);
+            doStreamWindow(authorization, traceId);
         }
 
         private void onStreamData(
@@ -787,7 +787,7 @@ public final class KafkaClientConnectionPool
             long authorization,
             long traceId)
         {
-            if (connection.initialBudId != NO_BUDGET_ID)
+            if (KafkaState.initialOpened(connection.state))
             {
                 final long initialSeqOffsetPeek = initialSeqOffset.peekLong();
 
@@ -817,8 +817,6 @@ public final class KafkaClientConnectionPool
 
             doBegin(sender, originId, routedId, replyId, replySeq, replyAck, replyMax,
                 traceId, authorization, connection.initialBudId, EMPTY_EXTENSION);
-
-            doStreamWindow(authorization, traceId);
         }
 
         private void doStreamData(
@@ -1080,6 +1078,13 @@ public final class KafkaClientConnectionPool
             if (KafkaState.closed(state))
             {
                 state = 0;
+                initialAck = 0;
+                initialSeq = 0;
+                initialMax = 0;
+                replyAck = 0;
+                replySeq = 0;
+                replyMax = 0;
+                initialBudId = NO_BUDGET_ID;
             }
 
             if (!KafkaState.initialOpening(state))
@@ -1577,7 +1582,7 @@ public final class KafkaClientConnectionPool
 
         private void cleanupBudgetCreditorIfNecessary()
         {
-            if (initialBudId != NO_CREDITOR_INDEX)
+            if (initialBudId != NO_BUDGET_ID)
             {
                 creditor.release(initialBudId);
                 initialBudId = NO_BUDGET_ID;

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -787,23 +787,26 @@ public final class KafkaClientConnectionPool
             long authorization,
             long traceId)
         {
-            final long initialSeqOffsetPeek = initialSeqOffset.peekLong();
-
-            if (initialSeqOffsetPeek != NO_OFFSET)
+            if (connection.initialBudId != NO_CREDITOR_INDEX)
             {
-                assert initialAck <= connection.initialAck - initialSeqOffsetPeek + initialAckSnapshot;
+                final long initialSeqOffsetPeek = initialSeqOffset.peekLong();
 
-                initialAck = connection.initialAck - initialSeqOffsetPeek + initialAckSnapshot;
-
-                if (initialAck == initialSeq)
+                if (initialSeqOffsetPeek != NO_OFFSET)
                 {
-                    initialSeqOffset.removeLong();
-                    initialAckSnapshot = initialAck;
-                }
-            }
+                    assert initialAck <= connection.initialAck - initialSeqOffsetPeek + initialAckSnapshot;
 
-            doWindow(sender, originId, routedId, initialId, initialSeq, initialAck, connection.initialMax,
-                traceId, authorization, connection.initialBudId, connection.initialPad);
+                    initialAck = connection.initialAck - initialSeqOffsetPeek + initialAckSnapshot;
+
+                    if (initialAck == initialSeq)
+                    {
+                        initialSeqOffset.removeLong();
+                        initialAckSnapshot = initialAck;
+                    }
+                }
+
+                doWindow(sender, originId, routedId, initialId, initialSeq, initialAck, connection.initialMax,
+                    traceId, authorization, connection.initialBudId, connection.initialPad);
+            }
         }
 
         private void doStreamBegin(

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -1083,7 +1083,6 @@ public final class KafkaClientConnectionPool
                 initialMax = 0;
                 replyAck = 0;
                 replySeq = 0;
-                replyMax = 0;
                 initialBudId = NO_BUDGET_ID;
             }
 

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -787,7 +787,7 @@ public final class KafkaClientConnectionPool
             long authorization,
             long traceId)
         {
-            if (connection.initialBudId != NO_CREDITOR_INDEX)
+            if (connection.initialBudId != NO_BUDGET_ID)
             {
                 final long initialSeqOffsetPeek = initialSeqOffset.peekLong();
 
@@ -1580,7 +1580,7 @@ public final class KafkaClientConnectionPool
             if (initialBudId != NO_CREDITOR_INDEX)
             {
                 creditor.release(initialBudId);
-                initialBudId = NO_CREDITOR_INDEX;
+                initialBudId = NO_BUDGET_ID;
             }
         }
 

